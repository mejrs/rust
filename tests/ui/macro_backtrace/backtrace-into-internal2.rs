//! We try to hide the internals of builtin-like macros from error messages,
//! but *not* if `-Z macro-backtrace` is enabled

//@ revisions: default with
//@[with] compile-flags: -Z macro-backtrace
//@[with] error-pattern: in this expansion of `$crate::format_args_nl!`

fn main(){
    let x: u32;

    println!("{x}");
    //~^ERROR used binding `x` isn't initialized
}
