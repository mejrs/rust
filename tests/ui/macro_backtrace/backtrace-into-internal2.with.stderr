error[E0381]: used binding `x` isn't initialized
  --> $DIR/backtrace-into-internal2.rs:11:15
   |
LL |     let x: u32;
   |         - binding declared here but left uninitialized
LL |
LL |     println!("{x}");
   |     ----------^^^--
   |     |         |
   |     |         `x` used here but it isn't initialized
   |     in this macro invocation (#1)
  --> $SRC_DIR/std/src/macros.rs:LL:COL
   |
   = note: in this expansion of `println!` (#1)
  ::: $SRC_DIR/std/src/macros.rs:LL:COL
   |
   = note: in this macro invocation (#2)
  --> $SRC_DIR/core/src/macros/mod.rs:LL:COL
   |
   = note: in this expansion of `$crate::format_args_nl!` (#2)
   |
help: consider assigning a value
   |
LL |     let x: u32 = 42;
   |                ++++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0381`.
